src/transformers/generation/utils.py

```diff
@@ -3769,6 +3642,153 @@ def _prefill_chunking(self, input_ids: torch.LongTensor, generation_config: Gene
+            else:
+                generate_output["scores"] = generate_output["scores"] + (score_tensors,)
+
+        if generate_output["output_logits"] and logits_tensors is not None:
+            if isinstance(logits_tensors, (tuple, list)):
+                generate_output["raw_logits"] = generate_output["raw_logits"] + tuple(logits_tensors)
+            else:
+                generate_output["raw_logits"] = generate_output["raw_logits"] + (logits_tensors,)
+
+        # Assisted path: append pre-split chunks (many per step)
+        if generate_output["output_attentions"]:
+            if decoder_attentions_chunks is not None:
+                generate_output["decoder_attentions"] = generate_output["decoder_attentions"] + tuple(
+                    decoder_attentions_chunks
+                )
+            if self.config.is_encoder_decoder and cross_attentions_chunks is not None:
+                generate_output["cross_attentions"] = generate_output["cross_attentions"] + tuple(
+                    cross_attentions_chunks
+                )
+
+        if generate_output["output_hidden_states"] and decoder_hidden_states_chunks is not None:
+            generate_output["decoder_hidden_states"] = generate_output["decoder_hidden_states"] + tuple(
+                decoder_hidden_states_chunks
+            )
+
+        # Standard path: single-step append from ModelOutput (**outputs)
+        # Prefer decoder_* keys when present; fall back to decoder-only keys.
+        if generate_output["output_attentions"]:
+            if self.config.is_encoder_decoder:
+                if decoder_attentions is not None:
+                    generate_output["decoder_attentions"] = generate_output["decoder_attentions"] + (
+                        decoder_attentions,
+                    )
+                if cross_attentions is not None:
+                    generate_output["cross_attentions"] = generate_output["cross_attentions"] + (cross_attentions,)
+            else:
+                if attentions is not None:
+                    generate_output["decoder_attentions"] = generate_output["decoder_attentions"] + (attentions,)
+
+        if generate_output["output_hidden_states"]:
+            if self.config.is_encoder_decoder:
+                if decoder_hidden_states is not None:
+                    generate_output["decoder_hidden_states"] = generate_output["decoder_hidden_states"] + (
+                        decoder_hidden_states,
+                    )
+            else:
+                if hidden_states is not None:
+                    generate_output["decoder_hidden_states"] = generate_output["decoder_hidden_states"] + (
+                        hidden_states,
+                    )
+
+    def _finalize_generate_output(
+        self,
+        generate_output: dict,
+        sequences,
+        model_kwargs,
+        decoder_only_cls=GenerateDecoderOnlyOutput,
+        encoder_decoder_cls=GenerateEncoderDecoderOutput,
```

Comment:
I'd prefer to have no defaults, so that the output type is clear in each decoding method